src/videojs-contrib-hls.js

```diff
@@ -674,13 +674,20 @@ const HlsSourceHandler = function(mode) {
 
       let settings = videojs.mergeOptions(options, {hls: {mode}});
 
+      let previousBeforeRequest;
+
+      if (tech.hls && tech.hls.xhr) {
```

Comment:
Not sure how I feel about this. We should be cleaning up `tech.hls` on dispose anyway. That we don't is actually a bug.